ownCloud/Settings/Passcode/PasscodeViewController.swift

```diff
@@ -257,4 +265,54 @@ class PasscodeViewController: UIViewController, Themeable {
 		cancelButton?.applyThemeCollection(collection, itemStyle: .defaultForItem)
 		cancelButton?.layer.cornerRadius = 0
 	}
+    
+    // MARK: - External Keyboard Commands
+    
+    @objc func performKeyCommand(sender: UIKeyCommand) {
+        guard let key = sender.input else {
+            return
+        }
+        
+        switch key {
+        case "\u{8}":
+            deleteLastDigit()
+        case UIKeyCommand.inputEscape:
+            cancelHandler?(self)
+        default:
+            appendDigit(digit: key)
+        }
+        
+    }
+    
+    override var keyCommands: [UIKeyCommand]? {
```

Comment:
@hosy why is it optional? Below you initialize an empty array which is always returned.. Also can we do lazy init here instead of creating new array whenever getter is called?